website/docs/tstruct.md

```diff
@@ -18,17 +18,19 @@ require 'sorbet-runtime'
 class MyStruct < T::Struct
   prop :x, Integer
   const :y, T.nilable(String)
+  const :z, Float, default: 0.5
 end
 ```
 
 This is basically the same as having written code like this:
 
 ```ruby
 class MyStruct < T::Struct
-  sig {params(x: Integer, y: String).void}
-  def initialize(x:, y: nil)
+  sig {params(x: Integer, y: String, z: Float).void}
+  def initialize(x:, y: nil, z: 0.5)
     @x = x
     @y = y
+    @z = z
```

Comment:
Thinking about this example, is it even accurate to show people these instance variables?

I don't think that's how the runtime works, and I don't think we want people to be able to know about T::Struct internals even if it does work this way. What do you think about eliding the method bodies of all the DSL-synthesized methods in this page?

Reply:
I agree that it might be good to not expose the inner workings of the runtime machinery by showing the bodies of the generated methods, especially if that's not actually the way they work. But having the bodies might help a user develop a mental model for how the generated methods work (ah, the initialize sets up the internal data the way I expect, and the getter and setter methods get and update the internal data accordingly). It could be argued, however, that even without the method bodies, it's pretty clear what's going on.

Reply:
Actually, one case I feel we should perhaps document explicitly is what the setter methods return. Do they return the old value, or the new value? Based on the current example, it looks like they return the new value - is that accurate?

Reply:
Update: I think I was misunderstanding how Ruby works. It seems that you can't make an `x=` method return anything other than the RHS.